### PR TITLE
Add support for workspace inheritance

### LINKED ISSUE #1180: `Error: failed to parse manifest` when using the workspace inheritance feature from Rust 1.64
## 🐛 Bug description
I set up the my `Cargo.toml` files to use the [workspace inheritance feature](https://blog.rust-lang.org/2022/09/22/Rust-1.64.0.html#cargo-improvements-workspace-inheritance-and-multi-target-builds) from Rust 1.64, which resulted in this error:

```
$ wasm-pack build --target no-modules --out-dir ../web/pkg
Error: failed to parse manifest: /Users/printfn/Code/fend/wasm/Cargo.toml
Caused by: invalid type: map, expected a string for key `package.description`
```

Compiling the same project with `cargo build` works fine, so this error must be specific to `wasm-pack`.

#### 🤔 Expected Behavior
I would've expected the Cargo.toml file to be parsed correctly.

#### 👟 Steps to reproduce
Workspace Cargo.toml:
```toml
[workspace]
members = ["cli", "core", "wasm"]

[workspace.package]
version = "1.1.0"
description = "Arbitrary-precision unit-aware calculator"
# ...
```

Wasm Cargo.toml:
```
[package]
name = "fend-wasm"
version.workspace = true
description.workspace = true
# ...
```

[This commit](https://github.com/printfn/fend/commit/25330f5649cb47a44b5778c75e34d6997a33bd3c) shows the diff between the `Cargo.toml` files.

#### 🌍 Your environment
Include the relevant details of your environment.
wasm-pack version: 0.10.3
rustc version: 1.64

```
$ cargo -vV
cargo 1.64.0 (387270bc7 2022-09-16)
release: 1.64.0
commit-hash: 387270bc7f446d17869c7f208207c73231d6a252
commit-date: 2022-09-16
host: aarch64-apple-darwin
libgit2: 1.4.2 (sys:0.14.2 vendored)
libcurl: 7.79.1 (sys:0.4.55+curl-7.83.1 system ssl:(SecureTransport) LibreSSL/3.3.6)
os: Mac OS 12.6.0 [64-bit]
```

### PR DESCRIPTION
**Pull request by printfn**
This PR adds support for reading metadata from crates using the [workspace inheritance](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table) feature introduced in [Rust 1.64](https://blog.rust-lang.org/2022/09/22/Rust-1.64.0.html#cargo-improvements-workspace-inheritance-and-multi-target-builds).

I've implemented this by bumping the `cargo_metadata` dependency to the latest version, and using its metadata for the `description`, `license`, `license-file`, `repository` and `homepage` fields, instead of parsing the `Cargo.toml` file directly. This should also make `wasm-pack` more resilient to future `Cargo.toml` format changes.

I've tested the change locally and it seems to work perfectly.

Fixes #1180.

---

Make sure these boxes are checked! 📦✅

- [x] You have the latest version of `rustfmt` installed
```bash
$ rustup component add rustfmt
```
- [x] You ran `cargo fmt` on the code base before submitting
- [x] You reference which issue is being closed in the PR text

✨✨ 😄 Thanks so much for contributing to wasm-pack! 😄 ✨✨

### CODE REVIEW

**Comment by haraldreingruber** [comment]
Anything I can help to move this PR forward?

**Comment by printfn** [comment]
It's blocked by https://github.com/rustwasm/wasm-pack/pull/1188

**Comment by haraldreingruber** [comment]
> It's blocked by #1188

Thanks for the update!

**Review by drager — APPROVED** [review]
Looks good and sounds like a better solution! Thanks a lot.

Will merge it once #1188 is merged.

**Comment by printfn** [comment]
Rebased onto #1188 and resolved the merge conflicts.

**Comment by yihau** [comment]
will this one release a new version? like v0.11.0 🤔

**Comment by drager** [comment]
> will this one release a new version? like v0.11.0 thinking

Yes, will do that in a near future :)

**Comment by yihau** [comment]
Hi there! Just would like to bump this up. How's the new release going? Anything I can help with it? can't wait to build our project with this feature 😎

**Pull request merged**
